### ISSUE TITLE
JSONString as mutation input

### ISSUE THREAD

**Issue by tom-zeit** [graphql-python/graphene#223]
Hello,
I'm trying to implement a mutation this way:

```
from graphene.core.types.custom_scalars import JSONString
class IntroduceRecord(graphene.Mutation):
    class Input:
        data = JSONString()

    record = graphene.Field(RecordNode)

    @classmethod
    def mutate(cls,instance, input, info):    
        data = input.get('data')
        return IntroduceRecord(record=RecordNode(models.Record.objects.all()[0]))
```

This is the mutation:
```
mutation insert {
    introduceRecord(data:"{\"bla\":\"5\"}") {
        record {
          uid
        }
    }
}
```

But in the mutate classmethod the data input is of type string instead of a dict.
Is it the intended behaviour? And if so, what is the need for JSONString?

This could be fixed by changing the parse_literal staticmethod in JSONString class from
```
   @staticmethod
    def parse_literal(node):
        if isinstance(node, ast.StringValue):
            return json.dumps(node.value)
```

to 

   ```
@staticmethod
    def parse_literal(node):
        if isinstance(node, ast.StringValue):
            return json.loads(node.value)
```

**Comment by saharhashai** [comment]
Happens here as well.

**Comment by syrusakbary** [comment]
Oh! Good catch! This is a unintended behaviour

**Comment by syrusakbary** [comment]
This is fixed in `master`. Closing the issue!

**Comment by joshourisman** [comment]
It appears that this fix may have caused a new bug with Relay mutations. Specifically, when I just updated my graphene install to 1.x (1.1.3, specifically), a mutation that had previously been working with a `JSONString()` in the input started failing. I was eventually able to determine that `ClientIDMutation.mutate_and_get_payload()` expects the input value to be a string, not an object.

I was able to resolve the issue and get my mutation working again by simply changing the `JSONString.parse_value()` method back to it's original form:

```
class JSONString(graphene.types.json.JSONString):
    @staticmethod
    def parse_value(value):
        return json.dumps(value)
```
